### ISSUE TITLE
ReferenceError: HTMLCanvasElement is not defined

### ISSUE THREAD

**Issue by USMortality** [chartjs/chartjs-plugin-annotation#686]
Hi, I'm getting an error when updating from version 1.1.0 to 1.2.0 or even latest version:

```
/Users/ben/dev/covid_charts/node_modules/chartjs-plugin-annotation/dist/chartjs-plugin-annotation.js:248
  return content instanceof Image || content instanceof HTMLCanvasElement;
                                                        ^

ReferenceError: HTMLCanvasElement is not defined
    at isImageOrCanvas (/Users/ben/dev/covid_charts/node_modules/chartjs-plugin-annotation/dist/chartjs-plugin-annotation.js:248:57)
    at measureLabelSize (/Users/ben/dev/covid_charts/node_modules/chartjs-plugin-annotation/dist/chartjs-plugin-annotation.js:277:7)
    at loadLabelRect (/Users/ben/dev/covid_charts/node_modules/chartjs-plugin-annotation/dist/chartjs-plugin-annotation.js:818:20)
    at LineAnnotation.resolveElementProperties (/Users/ben/dev/covid_charts/node_modules/chartjs-plugin-annotation/dist/chartjs-plugin-annotation.js:754:14)
    at updateElements (/Users/ben/dev/covid_charts/node_modules/chartjs-plugin-annotation/dist/chartjs-plugin-annotation.js:1570:27)
    at Object.afterUpdate (/Users/ben/dev/covid_charts/node_modules/chartjs-plugin-annotation/dist/chartjs-plugin-annotation.js:1478:5)
    at callback (/Users/ben/dev/covid_charts/node_modules/chart.js/dist/chart.js:800:15)
    at PluginService._notify (/Users/ben/dev/covid_charts/node_modules/chart.js/dist/chart.js:6187:11)
    at PluginService.notify (/Users/ben/dev/covid_charts/node_modules/chart.js/dist/chart.js:6174:25)
    at Chart.notifyPlugins (/Users/ben/dev/covid_charts/node_modules/chart.js/dist/chart.js:7312:26)
    at Chart.update (/Users/ben/dev/covid_charts/node_modules/chart.js/dist/chart.js:6902:10)
    at new Chart (/Users/ben/dev/covid_charts/node_modules/chart.js/dist/chart.js:6667:12)
    at ChartJSNodeCanvas.renderChart (/Users/ben/dev/covid_charts/node_modules/chartjs-node-canvas/dist/index.js:196:23)
    at ChartJSNodeCanvas.renderToBuffer (/Users/ben/dev/covid_charts/node_modules/chartjs-node-canvas/dist/index.js:79:28)
    at makeChart (file:///Users/ben/dev/covid_charts/dist/src/chart.js:184:36)
    at analyzeSeries (file:///Users/ben/dev/covid_charts/dist/src/main.js:25:25)
```

Any idea what's wrong?

**Comment by kurkle** [comment]
What version of typescript are you using?
I'd guess it has something to do with tsconfig.json

**Comment by USMortality** [comment]
Thanks. I'm using: `"typescript": "^4.5.5"`

Here's my tsconfig:
```

    "compilerOptions": {
        "outDir": "dist",
        "sourceMap": true,
        "target": "es2021",
        "module": "esnext",
        "moduleResolution": "node",
        "esModuleInterop": true,
        "skipLibCheck": true,
        "resolveJsonModule": true,
        "typeRoots": [
            "node_modules/@types",
            "types"
        ],
    },
    "include": [
        "src",
        "types",
        "test"
    ],
    "exclude": [
        "node_modules"
    ]
}
```

I have a branch with the issue up here, in case you want to look:
https://github.com/USMortality/COVID19-Charts/tree/chart-js-new
cmd: `npm run start "world" "total_cases_per_million" "covid_cases_million" "COVID-19 Cases (per Million)" '["germany"]'`

Any other thoughts? On how to debug?
The error happens right on the: `await chartJSNodeCanvas.renderToBuffer(configuration);` call

**Comment by kurkle** [comment]
Ok, its an overlook on the plugin. You are using chartjs-node-canvas and there is no DOM, which would provide those classes. I think you can work around it by just creatindg dummy classes by those names.

**Comment by stockiNail** [comment]
> I think you can work around it by just creating dummy classes by those names.

Does it mean these classes should be added to the plugin project?

**Comment by kurkle** [comment]
> > I think you can work around it by just creating dummy classes by those names.
> 
> Does it mean these classes should be added to the plugin project?

No, but we should detect if they exist before trying to use them.

**Comment by USMortality** [comment]
Thx. Yes that's what I understood too, because `HTMLCanvasElement` is only available really in the browser...

So I tried to create a custom class, but my TS compiler tells, me this class is already existing in the typescript package...
<img width="611" alt="Screen Shot 2022-02-14 at 10 28 26 AM" src="https://user-images.githubusercontent.com/94856880/153893713-55070251-e85f-4cdb-9b52-a4d788856b55.png">

/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib

I also tried to add this line to my tsconfig:
```
    "lib": [
        "dom",
    ]
```

Any other hints? Thanks.
